pylhc/calibration/dispersion.py

```diff
@@ -0,0 +1,277 @@
+    err = dispersion_err[dispersion_values.index.isin(positions.index)]
+
+    # Get the curve fit for the expected affine function
+    fit, fit_cov = curve_fit(dispersion_function, positions, values, sigma=err)
+
+    # Get the error from the covariance matrix
+    fit_err = np.sqrt(np.diag(fit_cov))
+
+    # Get the fitted beta and add the errors to get min/max values
+    dispersion_fit = dispersion_function(positions, fit[0], fit[1])
+    dispersion_max_fit = dispersion_function(
+        positions, fit[0] + fit_err[0], fit[1] + fit_err[1]
+    )
+    dispersion_min_fit = dispersion_function(
+        positions, fit[0] - fit_err[0], fit[1] - fit_err[1]
+    )
+    dispersion_fit_err = (dispersion_max_fit - dispersion_min_fit) / 2
+
+    return dispersion_fit, dispersion_fit_err
+
+
+def _get_factors_from_dispersion(
+    dispersion: Dict[str, pd.Series]
+) -> Tuple[pd.Series, pd.Series]:
+    """
+    This function computes the calibration factors for the dispersion method
+    with the non fitted dispersion values. The associated error is also
+    calculated.
+
+    Args:
+      dispersion (Dict[str, pd.Series]): Dictionnary containing 4 keys: phase,
+        phase_err, amp and amp_err. Each key is related to the method used to
+        obtain the dispersion and its error.
+
+    Returns:
+      Tuple[pd.Series, pd.Series]: The first Series are the calibration
+        factors, the second one their error.
+    """
+    # Get the ratios, those are our calibration factors
+    factors = dispersion["phase"] / dispersion["amp"]
+
+    # Code in BBs
+    calibration_error = (dispersion["phase_err"] / dispersion["amp"]) ** 2
+    calibration_error += (
+        dispersion["amp_err"] * (dispersion["phase"] / (dispersion["amp"] ** 2))
+    ) ** 2
+    calibration_error = np.sqrt(calibration_error)
+
+    return factors, calibration_error
+
+
+def _get_factors_from_dispersion_fit(
+    dispersion: Dict[str, pd.Series]
+) -> Tuple[pd.Series, pd.Series]:
+    """
+    This function computes the calibration factors for the dispersion method
+    with the _fitted_ dispersion values. The associated error is also
+    calculated.
+
+    Args:
+      dispersion (dict): Dictionnary containing 4 keys: phase_fit,
+      phase_fit_err, amp and amp_err.  Each key is related to the method used
+      to obtain the dispersion and its error.
+
+    Returns:
+      Tuple[pd.Series, pd.Series]: The first Series are the calibration
+      factors, the second one their error.
+    """
+    # Get the ratios, those are our calibration factors
+    factors = dispersion["phase_fit"] / dispersion["amp"]
+
+    calibration_error = (dispersion["phase_fit_err"] / dispersion["amp"]) ** 2
+    calibration_error += (
+        dispersion["amp_err"] * dispersion["phase_fit"] / (dispersion["amp"] ** 2)
+    ) ** 2
+    calibration_error = np.sqrt(calibration_error)
+
+    return factors, calibration_error
```

Comment:
Similar to beta...Practically identical methods...

Reply:
also merged